pkg/core/console/adjudicate.go

```diff
@@ -0,0 +1,163 @@
+
+	endpointsResp, err := cs.eth.GetRegisteredEndpointsForServiceProvider(
+		ctx,
+		connect.NewRequest(&ethv1.GetRegisteredEndpointsForServiceProviderRequest{Owner: serviceProviderAddress}),
+	)
+	if err != nil {
+		cs.logger.Error("Falled to get service provider endpoints", "address", serviceProviderAddress, "error", err)
+		return err
+	}
+	endpoints := endpointsResp.Msg.Endpoints
+
+	// Get total number of cometbft validators in order to calculate SLA performance later
+	totalValidators, err := cs.db.TotalValidators(ctx)
+	if err != nil {
+		cs.logger.Error("Falled to get total validators", "error", err)
+		return err
+	}
+
+	// configure start and end times
+	startTime := time.Now().Add(-7 * 24 * time.Hour)
+	endTime := time.Now()
+	if c.QueryParam("start") != "" {
+		if parsed, err := time.Parse("2006-01-02T15:04", c.QueryParam("start")); err != nil {
+			cs.logger.Warn("failed to parse start time from query string", "error", err)
+		} else {
+			startTime = parsed
+		}
+	}
+	if c.QueryParam("end") != "" {
+		if parsed, err := time.Parse("2006-01-02T15:04", c.QueryParam("end")); err != nil {
+			cs.logger.Warn("failed to parse end time from query string", "error", err)
+		} else {
+			endTime = parsed
+		}
+	}
+
+	// Populate endpoints and their SLAs for the view model
+	viewEndpoints := make([]*pages.Endpoint, len(endpoints))
+	totalMetSlas, totalPartialSlas, totalDeadSlas := 0, 0, 0
+	for i, ep := range endpoints {
+		viewEndpoints[i] = &pages.Endpoint{
+			Endpoint:   ep.Endpoint,
+			EthAddress: ep.DelegateWallet,
+		}
+		if err != nil && !errors.Is(err, pgx.ErrNoRows) {
+			cs.logger.Error("Falled to get rollups in time range", "start time", startTime, "end time", endTime, "error", err)
+			return err
+		}
+
+		var cometAddress string
+		validator, err := cs.db.GetNodeByEndpoint(ctx, ep.Endpoint)
+		if err != nil && !errors.Is(err, pgx.ErrNoRows) {
+			cs.logger.Error("Falled to get cometbft validator for endpoint", "endpoint", ep.Endpoint, "error", err)
+			return err
+		} else if !errors.Is(err, pgx.ErrNoRows) {
+			cometAddress = validator.CometAddress
+		}
+		viewEndpoints[i].CometAddress = cometAddress
+
+		slaRollups, err := cs.db.GetRollupReportsForNodeInTimeRange(
+			ctx,
+			db.GetRollupReportsForNodeInTimeRangeParams{
+				Address: cometAddress,
+				Time:    cs.db.ToPgxTimestamp(startTime),
+				Time_2:  cs.db.ToPgxTimestamp(endTime),
+			},
+		)
+		if err != nil && !errors.Is(err, pgx.ErrNoRows) {
+			cs.logger.Error("Failed to get rollups from db for node", "address", cometAddress, "start time", startTime, "end time", endTime, "error", err)
+			return err
+		}
+
+		// Calculate the status of each SLA rollup as met, partially met, or dead (0% met)
+		viewSlaReports := make([]*pages.AdjudicateSlaReport, len(slaRollups))
+		for i, r := range slaRollups {
+			reportQuota := int32(0)
+			if totalValidators > 0 {
+				reportQuota = int32(r.BlockEnd-r.BlockStart) / int32(totalValidators)
+			}
+
+			var status pages.SlaStatus
```

Comment:
prefer we get rid of partial statuses for now. make this as cut & dry as possible